linebot/filter.go

```diff
@@ -0,0 +1,327 @@
+// Copyright 2020 LINE Corporation
+//
+// LINE Corporation licenses this file to you under the Apache License,
+// version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at:
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package linebot
+
+import "encoding/json"
+
+// Filter interface
+type Filter interface {
+	Filter()
+}
+
+// AudienceFilter type
+type AudienceFilter struct {
```

Comment:
The Audience object is not an demographic filter object, it is a recipient object.
Currently recipient object and demographic filter object have (looks) same operator object but they are not exactly same, they can be altered for each.
So I think it is better to split operator struct to two, 1. recipient operator object 2. demographic filter operator object.

Reply:
That's a good idea, will update later.

Reply:
Hi, I've renamed `filter` to `selector` to remedy the semantics.  Thanks.